File: Mudit Choraria/sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int n = size(nums);
        int l = 0, r = n - 1;
        while(l < r) {
            if(nums[l] % 2 && nums[r] % 2 == 0) {
                swap(nums[l], nums[r]);
            }
            if(nums[l] % 2 == 0) {
                l++;
            }
            if(nums[r] % 2) {
                r--;
            }
        }
        return nums;
    }
};

File: Mahesh_Babar/sort_array_by_parity.java
// https://leetcode.com/problems/sort-array-by-parity/

class sort_array_by_parity {
    public int[] sortArrayByParity(int[] nums) {
        int i = 0;
        int j = nums.length - 1;

        while (i < j) {
            if (nums[i] % 2 > nums[j] % 2) {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
            }
            if (nums[i] % 2 == 0)
                i++;
            if (nums[j] % 2 != 0)
                j--;
        }
        return nums;
    }
}
